controller/src/main/java/io/pravega/controller/store/stream/ZKStoreHelper.java

```diff
@@ -177,13 +177,17 @@ public ZKStoreHelper(final CuratorFramework cf, Executor executor) {
     }
 
     CompletableFuture<List<String>> getChildren(final String path) {
+        return getChildren(path, true);
+    }
+
+    CompletableFuture<List<String>> getChildren(final String path, boolean ignoreDataNotFound) {
```

Comment:
I don't understand the purpose of this flag. Why would we want to ignore data not found sometimes?

Reply:
Earlier, for getChildren dataNotFound was translated into empty list by default. 
Everywhere else in zookeeperstreammetadatastore we did getChildren and expected empty list if there are no children or if the parent didn’t exist. However that was detrimental in determining if scope didn’t exist while doing list stream on scope for zookeeper based store. Hence I added a new api which did not translate the data not found to empty list

Reply:
I understand that when listing streams you want the `StoreException.DataNotFoundException` to be thrown, but I don't understand why we need to allow two behaviors, that's confusing. What prevents us from always propagating `StoreException.DataNotFoundException` instead of having a dual behavior?

Reply:
So getChildren is used in two places - get txns in epoch and get streams in scope. 
We had implemented the getChildren method to expect data not found and return empty list. 
We can have a single behaviour for getChildren as long as that is throwing data not found exception. 

And if we do that, we will have to handle that exception in transactionsInEpoch call to retain existing behaviour of that api. 
I thought instead of disturbing existing code, I would rather add a new getChildren api that threw the desired exception

Reply:
I suspect that the change in `transactionsInEpoch` to handle the exception should be minimal, but you tell me. If my suspicion is right, then I'd rather make that change and make the behavior consistent. Another option is to merge this one as is and create another issue to remove the flag and make the behavior consistent, but I'd want to do it right away to avoid the risk of releasing with this flag.

Given the size of this PR and the necessary changes to make it consistent, we might as well do it here.

Reply:
Actually i checked where all `getChildren` is called:
1. ZkOrderer : 4 different places
2. ZKScope: listStream
3. ZkStream: 5 different methods - all related to transactions
4. ZkStreamMetadataStore:  getCompletedTxn, listScopes
5. ZkBucketStore: getStreamsInBucket

So it will be a substantial change